### DISCUSSION TITLE
Input readonly, custom input placeholder, Input width and many other

### DISCUSSION THREAD

**Discussion by ziaKhan1995** [stephanrauh/ngx-extended-pdf-viewer#2330] [Ideas]
@stephanrauh Thank you for this wonderful library. 

### We are looking forward to some enhancements in the upcoming version:

Implement read-only functionality for the input field.
Customize placeholder text for the input field to specify its purpose, such as requesting someone to sign the document with details like company name, email address, etc.

1. Introduce the ability to insert editor annotations with customizable input fields, approximately 200px in size.
2. Include a reference to the input field for better accessibility and usability.
3. Incorporate checkbox/radio button functionality, similar to how we handle text or image insertion.
4. These are just a few among several other features we are considering.
5. And how to add custom HTML as editor annotation which we can control from our own html/ts component.

We have observed the UI in **app.box.com**, particularly when selecting the **Sign Myself** option for document signing. It's noteworthy how seamlessly it integrates with various input types:

Here is app.box.com UI when we **Sign Myself** the document
![image](https://github.com/stephanrauh/ngx-extended-pdf-viewer/assets/77499979/8ca49831-e452-4f49-8c5c-872f4e3c2778)

For instance, when selecting **company** for **signing**, the input field automatically populates with a **placeholder for the company** name. Similarly, when opting for **Email signing**, the input field displays a default email value, albeit@gmail.com in a **read-only state**. Furthermore, selecting Date inserts the current date into the input field, also set to read-only.

We are curious if it's feasible to incorporate similar functionality into ngx-extended-pdf-viewer in the upcoming version.

**Comment by stephanrauh** [comment]
My answer consists of two parts:

1. I'm afraid I don't know how to add custom HTML to the PDF document,
2. but I can tell you how to manipulate input fields that are already part of the PDF document.

As for 1: adding custom HTML is a major research project (at least to me). The PDF viewer implements most of the standard annotations of the PDF standard and stored them in `PDFViewerApplication.pdfDocument.annotationStorage`. It should be possible to reverse-engeneer the data structure and to find out how to re-render the annotation layer renderer. But at the moment, I can't tell you much more. Can I talk you into finding out yourself? That could speed up things considerably.

As for 2: Capture the `(annotationLayerRendered)` event. It's called after creating the HTML element of a single page. You can safely manipulate the input fields in this event. This way, you can implement things like enabling/disabling the field or adding placeholders.

> Include a reference to the input field for better accessibility and usability.

I'm not sure what you definition of "reference" is - but if it is something like the `id` or `name` attribute, I'm afraid I can't help you. These two fields are generated deep inside pdf.js and they seem to be defined by / determined by the PDF document itself. More likely than not I'd break stuff by changing these attributes.
